### PR TITLE
More inline merge work

### PR DESCRIPTION
**Pull request by martinal**
Allow multiple notebook filenames to nbshow.

Remove trailing newline in inline source conflict.

Add some slightly more complex conflicting notebooks to test suite.
Will open an issue where these trigger bugs in output strategies.

### CODE REVIEW

File: nbdime/merging/decisions.py [hunk 1]
```diff
@@ -497,31 +497,29 @@ def _merge_tree(tree, sorted_paths):
...
-            if nextPath is not None:
-                trunk = push_path(path[len(nextPath):], trunk)
-                root = nextPath
+            trunk = push_path(path[len(nextPath):], trunk)
+            root = nextPath
         else:
             # We have started on a new trunk
             # Collect branches on the new trunk, and merge the trunks
-            newTrunk = _merge_tree(tree, sorted_paths[i + 1])
+            newTrunk = _merge_tree(tree, sorted_paths[i + 1:])
```
Comment (martinal) [inline nbdime/merging/decisions.py]: @vidartf is the change on this line right? Output clearing seems to work now.

**Comment by martinal** [comment]
Last push here fixes https://github.com/jupyter/nbdime/issues/186

File: nbdime/merging/decisions.py (hunk 1 above)
Comment (vidartf) [inline nbdime/merging/decisions.py]: Yes, this was an error in translation from typescript code. Nice catch ;)

**Comment by martinal** [comment]
Try e.g.  `nbmerge -o merged.ipynb --output-strategy=inline nbdime/tests/files/mixed-conflicts--*`

File: nbdime/merging/autoresolve.py [hunk 2]
```diff
@@ -112,22 +129,32 @@ def make_inline_outputs_value(base, le, re,
...
     outputs.append(output_marker(sep2))
     outputs.extend(remote)
     outputs.append(output_marker(sep3))
 
-    return outputs
+    # DEBUGGING:
+    if pre:
+        pre = [output_marker("BEGIN PRE")] + pre + [output_marker("END PRE")]
+    if post:
+        post = [output_marker("BEGIN POST")] + post + [output_marker("END POST")]
```
Comment (vidartf) [inline nbdime/merging/autoresolve.py]: Just a note to remove/comment out debugging code before merging

File: nbdime/merging/autoresolve.py [hunk 3]
```diff
@@ -112,22 +129,32 @@ def make_inline_outputs_value(base, le, re,
...
 
+    # Remove all from base
+    begin = 0
+    end = len(base)
+
+    # Return range to replace with marked up lines
+    return begin, end, inlined
+
+
+# TODO: Currently unused, remove?
```
Comment (vidartf) [inline nbdime/merging/autoresolve.py]: Sure, we'll have it in the git log for future reference

File: nbdime/merging/autoresolve.py (hunk 2 above)
Comment (martinal) [inline nbdime/merging/autoresolve.py]: Good catch...

**Pull request merged**